src/engine/options_finder.rs

```diff
@@ -10,17 +10,23 @@ pub struct OptionsFinder {
     result: TabryResult,
 }
 
+#[derive(PartialEq, Eq, Hash)]
+pub struct OptionResult {
+    pub value: String,
+    pub desc: Option<String>,
+}
+
 pub struct OptionsResults {
     prefix: String,
-    pub options: HashSet<String>,
+    pub options: HashSet<OptionResult>,
     pub special_options: HashSet<String>,
 }
 
 impl OptionsResults {
-    fn insert(&mut self, value: &str) {
+    fn insert(&mut self, value: &str, desc: Option<String>) {
         if value.starts_with(&self.prefix) {
             // TODO get_or_insert_owned() in nightly would be ideal
-            self.options.insert(value.to_owned());
+            self.options.insert(OptionResult { value: value.to_owned(), desc });
```

Comment:
Cloning desc in the caller instead of here means a clone happens regardless of whether `value.starts_with(&self.prefix)`. It would be better to pass in a reference and do the clone here. In this case, probably passing in `&s.description` (i.e. the parameter would be `desc: &Option<String>`) would work.

It's a pity we have to clone the string here at all even if we aren't going to eventually show description (e.g. bash),. It could probably be avoided by passing in a parameter like `needs_description`. But that's modifying multiple levels of functions, and anyway, in the future, I may try to fix all this code so that we don't have to clone anything (if I can get the lifetimes to co-operate). For now, could you add a comment here like

```
// TODO: don't clone description here if we don't need it (probably will get fixed if we can not clone any part of the config) 
```